vertx-sql-client/src/main/java/io/vertx/sqlclient/impl/SqlConnectionFactoryBase.java

```diff
@@ -75,35 +80,79 @@ protected SqlConnectionFactoryBase(VertxInternal vertx, SqlConnectOptions option
     this.netClient = vertx.createNetClient(netClientOptions, clientCloseFuture);
   }
 
+  private int advanceNextHost() {
+    return lastLiveHost.getAndIncrement() % socketAddresses.size();
+  }
+
+  private static class AddressesIterator implements Iterator<SocketAddress> {
```

Comment:
Could be enhanced in future to perform iteration based on specific order i.e. emulate pg jdbc `?targetServerType=preferSlave`